txnsync/mainloop.go

```diff
@@ -227,7 +227,7 @@ func (s *syncState) onTransactionPoolChangedEvent(ent Event) {
 	newBeta := beta(ent.transactionPoolSize)
 
 	// see if the newBeta is at least 10% smaller or bigger than the current one
-	if (float32(s.lastBeta)*(1.0-betaGranularChangeThreshold)) <= float32(newBeta) || (float32(s.lastBeta)*(1.0+betaGranularChangeThreshold)) >= float32(newBeta) {
+	if (float32(s.lastBeta)*(1.0+betaGranularChangeThreshold)) <= float32(newBeta) || (float32(s.lastBeta)*(1.0-betaGranularChangeThreshold)) >= float32(newBeta) {
```

Comment:
I think this is still wrong. Now it will never go into the if condition. 
You need to change the || to &&.

Reply:
Yes, you are right. I will add these changes in a separate PR (with a unit test) so that this PR will just contain the end to end test.